### DISCUSSION TITLE
How can I update the name of an Armorstand with packets?

### DISCUSSION THREAD

**Discussion by FloGamerMC** [Minestom/Minestom#824] [Help] [answered]
I want to update the name of an Armorstand with packets, because each player should have a outher name.

**Comment by ZakShearman** [comment, marked as answer]
You're going to want to modify the EntityMetaDataPacket and send it yourself inside the Entity#updateNewViewer method and anywhere else necessary for your use case.

My entities in my TowerDefence project have per-player display names and I override Entity#updateNewViewer and manually handle when the name changes by calling my updateCustomName method which sends a modified EntityMetaDataPacket to all necessary players.

Example code: [updateNewViewer](https://github.com/ZakShearman/mc-tower-defence/blob/main/src/main/java/pink/zak/minestom/towerdefence/model/mob/living/LivingEnemyMob.java#L233) [updateCustomName](https://github.com/ZakShearman/mc-tower-defence/blob/main/src/main/java/pink/zak/minestom/towerdefence/model/mob/living/LivingEnemyMob.java#L247)

As a sidenote, I'm not aware of a way you can do this with or in combination with the existing metadata as you can't intercept the createMetadataPacket whilst being player aware. This means you do have to manually override all ways of the metadata changing  you want to support - essentially disabling the entity's normal meatadata object usage.

**Reply by FloGamerMC** [reply]
Thx

**Reply by emortaldev** [reply]
Is it not `updateNewViewable`? I never properly know the difference

**Reply by ZakShearman** [reply]
There is no updateNewViewable method, it is updateNewViewer. That method is responsible for sending the entity meta to a new viewer.

**Reply by emortaldev** [reply]
Oops it's viewableRule that I got it confused with
